component/main.jsonnet

```
@@ -20,6 +21,82 @@ local namespace = kube.Namespace(params.namespace) {
   },
 };
 
+local cronJob = kube.CronJob('paralus-pgdb-log-cleanup-cronjob') {
+  metadata+: {
+    namespace: inv.parameters.paralus.namespace,
+  },
+  spec+: {
+    schedule: params.dbCronJob.schedule,
+    failedJobsHistoryLimit: params.dbCronJob.jobHistoryLimit.failed,
+    successfulJobsHistoryLimit: params.dbCronJob.jobHistoryLimit.successful,
+    jobTemplate+: {
+      spec+: {
+        template+: {
+          spec+: {
+            restartPolicy: 'OnFailure',
+            containers: [
+              {
+                name: 'postgres-cleanup',
+                image: '%(registry)s/%(repository)s:%(tag)s' % params.images.dbCronJob,
+                args: [
+                  'psql',
+                  '-h',
+                  '$(DB_ADDR)',
+                  '-U',
+                  '$(DB_USER)',
+                  '-d',
+                  '$(DB_NAME)',
+                  '-c',
+                  "DELETE FROM audit_logs WHERE time < NOW() - INTERVAL '%d days';" % params.dbCronJob.retention,
+                ],
+                env: [
+                  {
+                    name: 'PGPASSWORD',
+                    valueFrom: {
+                      secretKeyRef: {
+                        name: 'paralus-db',
+                        key: 'DB_PASSWORD',
+                      },
+                    },
+                  },
+                  {
+                    name: 'DB_ADDR',
+                    valueFrom: {
+                      secretKeyRef: {
+                        name: 'paralus-db',
+                        key: 'DB_ADDR',
+                      },
+                    },
+                  },
+                  {
+                    name: 'DB_USER',
+                    valueFrom: {
+                      secretKeyRef: {
+                        name: 'paralus-db',
+                        key: 'DB_USER',
+                      },
+                    },
+                  },
+                  {
+                    name: 'DB_NAME',
+                    valueFrom: {
+                      secretKeyRef: {
+                        name: 'paralus-db',
+                        key: 'DB_NAME',
+                      },
+                    },
+                  },
+                ],
```

This will create the hostname as `hostname:port` which is not supported by psql. Port needs to be specified with -p option. Alternatively the DSN which is already present in the secret can be used:

```suggestion
                  '$(DSN)',
                  '-c',
                  "DELETE FROM audit_logs WHERE time < NOW() - INTERVAL '%d days';" % params.dbCronJob.retention,
                ],
                env: [
                  {
                    name: 'DSN',
                    valueFrom: {
                      secretKeyRef: {
                        name: 'paralus-db',
                        key: 'DSN',
                      },
                    },
                  },
                ],
```